config/options.go

```diff
@@ -319,26 +312,24 @@ func optionsFromViper(configFile string) (*Options, error) {
 	// start a copy of the default options
 	o := NewDefaultOptions()
 	// New viper instance to save into Options later
-	v := viper.New()
+	// v := viper.New()
 	// Load up config
-	err := bindEnvs(o, v)
+	err := bindEnvs(o, o.viper)
 	if err != nil {
 		return nil, fmt.Errorf("failed to bind options to env vars: %w", err)
 	}
 
 	if configFile != "" {
-		v.SetConfigFile(configFile)
-		if err := v.ReadInConfig(); err != nil {
+		o.viper.SetConfigFile(configFile)
+		if err := o.viper.ReadInConfig(); err != nil {
 			return nil, fmt.Errorf("failed to read config: %w", err)
 		}
 	}
 
-	if err := v.Unmarshal(&o); err != nil {
+	if err := o.viper.Unmarshal(o); err != nil {
```

Comment:
I think this is actually the cause of the intermittent failure, and why a new viper instance is used, rather than the embedded one.  If viper unmarshals from inside the struct, it can overwrite itself before complete, and that can screw up the rest of the config handling.

Reply:
Hmm, I reverted this part of change but it looks like the intermittent test failure is still here.

Reply:
Finally figured out where the race condition came from. The test is no longer flaky.